### PR TITLE
Typescript transition 

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] desktop hot (reload of tsx works)
- [x] desktop cold
- [x] desktop prod
- [x] ios
- [x] android
- [x] jest
- [x] storyshots
- [x] storybook
- [x] rn-storybook
- [x] no flow in ci
- [x] no flow in precommit
- [x] tsc in precommit
- [x] tslint (just using babel-eslint for now)
- [x] prettier to minimize

### CODE REVIEW

**Comment by chrisnojima** [comment]
@MarcoPolo pushing fixes here

**Comment by chrisnojima** [comment]
- [ ] osx prod build
- [ ] ios prod build
- [ ] android prod build

@keybase/react-hackers this is ready

File: shared/styles/index.types.d.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
+// Styles from our designers
+import * as CSS from './css'
+export {default as globalColors} from './colors'
+
+export declare const transition: (...properties: Array<string>) => Object
+
+type _fakeFontDefSeeCommentsOnThisStyle = {
```
Comment (buoyad) [inline shared/styles/index.types.d.ts]: either add comment or rename type

Comment (chrisnojima) [reply shared/styles/index.types.d.ts]: done

**Pull request merged**